02.calendar/cal.rb

```diff
@@ -0,0 +1,24 @@
+require 'date'
+require 'optparse'
+
+params = ARGV.getopts("y:","m:")
+
+year = (params["y"] || Date.today.year).to_i # str型だとDate.new出来ないので
+month = (params["m"] || Date.today.month).to_i
+
+start_day = Date.new(year,month,1) # カレンダーで最初に表示する日
+end_day = Date.new(year,month,-1) # カレンダーで最後に表示する日
+
+# カレンダーのスタートは日曜日から始まり、最初に改行なしの空白を差し込めば、その曜日から出力を行える
+calendar_of_first_weeks_spaces = "   " * start_day.wday
+
+puts "#{month}月 #{year}".center(20)
+puts " 日 月 火 水 木 金 土"
```

Comment:
実際のcalコマンドだと、

```suggestion
puts "日 月 火 水 木 金 土"
```

のように、最初のスペースはなかったりしませんか？
そこまで再現できるとベストです。